lib/opentelemetry-proto/src/convert.rs

```diff
@@ -57,6 +64,159 @@ impl ResourceLogs {
     }
 }
 
+impl ResourceMetrics {
+    pub fn into_event_iter(self) -> impl Iterator<Item = Event> {
+        let resource = self.resource.clone();
```

Comment:
```suggestion
        let resource = self.resource.as_ref();
```

It should be possible to clone when we create the Vector event.

Reply:
I tried this, and it's a bit hard to manage due to lifetime restrictions

Reply:
we can prob switch to using Arc to save on cloning